src/Locations.ts

```diff
@@ -14,9 +18,11 @@ export default class Locations extends EventEmitter {
     const { location } = message.data;
     const member = this.space.getMemberFromConnection(message.connectionId);
 
-    const previousLocation = member.location;
-    member.location = location;
-    this.emit('locationUpdate', { member, currentLocation: location, previousLocation });
+    if (member) {
```

Comment:
OOI, under what conditions we get a presence update for a member that is not there?

Reply:
I think if the connection drops but they are connected on a different connection to the one the message came from,

```
  getMemberFromConnection(connectionId: string) {
    return this.members.find((m) => m.connections.includes(connectionId));
  }
```

where `connectionId` is taken from the Presence Message.

Could return `undefined`.

Generally though this is just to make TypeScript happy.